src/yival/finetune/sft_trainer.py

```diff
@@ -0,0 +1,180 @@
+"""
+This module provides an implementation of Supervised Fine-tuning trainer.
+
+"""
+import os
+
+import bitsandbytes as bnb  # type: ignore
+import torch
```

Comment:
If the environment of machine learning is optional, this import should be placed in functions, not modules.

In this case, if the class must be exported, then `import torch` should be placed in `SFTTrainer.__init__`, or inner functions.

Reply:
I put these import in inner functions , but they look really strange
<img width="844" alt="image" src="https://github.com/YiVal/YiVal/assets/55043304/c996fd76-8b36-44bd-97c9-4c75cc5167bc">

I'm new bird in python  project , Is this the right way ?

Reply:
This way is not recommanded by default, but exactly working for this case. An optional dependency needs to be imported when invoked.